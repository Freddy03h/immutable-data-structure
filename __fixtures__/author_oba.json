{
  "id": "17491b54-580f-4d5f-94e5-9b9e501871c4",
  "name": "ÅŒba",
  "first_name": "Tsugumi",
  "tasks": [
    {
      "id": "5195c5d1-9d3e-4a6c-9b29-077fc217d224",
      "series_id": "fc34ad95-27cf-4cba-aafc-e9c37de532b4",
      "author_id": "17491b54-580f-4d5f-94e5-9b9e501871c4",
      "job_id": "bf39f3fe-ce75-48b0-b557-877da935633b",
      "series": {
        "id": "fc34ad95-27cf-4cba-aafc-e9c37de532b4",
        "title": "Platinum End",
        "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d",
        "editions": [
          {
            "id": "73173187-5965-451c-adbe-84e3e719abc0",
            "title": "Edition simple",
            "series_id": "fc34ad95-27cf-4cba-aafc-e9c37de532b4",
            "publisher_id": "122e0ec3-f072-4c5d-b40d-3eba4a82fe1e",
            "volumes": [
              {
                "id": "27e50b0a-785e-4298-af2f-ae492c57c8a6",
                "title": "Tome 6",
                "number": 6,
                "date_parution": "2017-09-20",
                "img": "",
                "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
                "isbn": "9782820329011",
                "asin": "2820329012"
              },
              {
                "id": "f6f21458-e707-40ba-b9ec-f7f28d7c6343",
                "title": "Tome 5",
                "number": 5,
                "date_parution": "2017-05-31",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/61CV%2BQwDT7L.jpg",
                "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
                "isbn": "9782820328533",
                "asin": "2820328539"
              },
              {
                "id": "06fe038b-8c0d-4767-9167-ae3ca9bc3067",
                "title": "Tome 4",
                "number": 4,
                "date_parution": "2017-03-08",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51tKK5n%2B-3L.jpg",
                "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
                "isbn": "9782820328175",
                "asin": "2820328172"
              },
              {
                "id": "7c0ad1a9-4a80-4710-b11b-eed249456924",
                "title": "Tome 3",
                "number": 3,
                "date_parution": "2016-11-23",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51jAjenb0EL.jpg",
                "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
                "isbn": "9782820325402",
                "asin": "2820325408"
              },
              {
                "id": "975e5e39-8a47-4e0c-9a60-4a6256163257",
                "title": "Tome 2",
                "number": 2,
                "date_parution": "2016-09-07",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51sgTWp8W-L.jpg",
                "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
                "isbn": "9782820324948",
                "asin": "2820324940"
              },
              {
                "id": "04754705-044e-4292-8ed2-eee50ee91d2d",
                "title": "Tome 1",
                "number": 1,
                "date_parution": "2016-05-25",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51TgELYk1WL.jpg",
                "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
                "isbn": "9782820324603",
                "asin": "2820324606"
              }
            ]
          }
        ]
      }
    },
    {
      "id": "0b6a3f23-ed97-46a6-b8bb-25ced41d9442",
      "series_id": "277484f9-d839-4b44-9f43-0743e6db540f",
      "author_id": "17491b54-580f-4d5f-94e5-9b9e501871c4",
      "job_id": "bf39f3fe-ce75-48b0-b557-877da935633b",
      "series": {
        "id": "277484f9-d839-4b44-9f43-0743e6db540f",
        "title": "Bakuman",
        "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d",
        "editions": [
          {
            "id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
            "title": "Edition simple",
            "series_id": "277484f9-d839-4b44-9f43-0743e6db540f",
            "publisher_id": "4c9547ff-2ef6-439a-80b8-ea705a385b76",
            "volumes": [
              {
                "id": "2c7fb5f7-8e3c-4cd2-83f3-b2182aaa3155",
                "title": "Tome 20",
                "number": 20,
                "date_parution": "2014-11-28",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/61um-EvJIXL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505060468",
                "asin": "2505060460"
              },
              {
                "id": "f7064665-b7a8-4ece-bf9c-38fa2ed50159",
                "title": "Tome 19",
                "number": 19,
                "date_parution": "2014-07-04",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51lwosrDNBL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505060451",
                "asin": "2505060452"
              },
              {
                "id": "0a5d6649-4cd7-4bc0-ac56-f4167064b6e4",
                "title": "Tome 18",
                "number": 18,
                "date_parution": "2014-04-04",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51NOi76qm7L.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505060444",
                "asin": "2505060444"
              },
              {
                "id": "1238dea9-842e-4c09-8d5a-2d738b64650a",
                "title": "Tome 17",
                "number": 17,
                "date_parution": "2014-01-24",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51WX33roerL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505060062",
                "asin": "2505060061"
              },
              {
                "id": "ef30e49c-46ad-40db-a910-03ec06f3da86",
                "title": "Tome 16",
                "number": 16,
                "date_parution": "2013-10-18",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/61Q4SMRrLlL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505017295",
                "asin": "2505017298"
              },
              {
                "id": "de6cb0d3-d128-49a5-a410-8fbd8c51bb49",
                "title": "Tome 15",
                "number": 15,
                "date_parution": "2013-07-05",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/510bEcHF6QL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505017288",
                "asin": "250501728X"
              },
              {
                "id": "65435d42-5c34-4359-a776-ea4eb00be938",
                "title": "Tome 14",
                "number": 14,
                "date_parution": "2013-04-05",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51-EE9Xj0fL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505017271",
                "asin": "2505017271"
              },
              {
                "id": "bc78b63f-e602-453f-b75d-f91a92416114",
                "title": "Tome 13",
                "number": 13,
                "date_parution": "2013-01-04",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51Tt6c27sWL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505016847",
                "asin": "2505016844"
              },
              {
                "id": "19195ca1-dee9-48a7-8c39-9906a1f2b0ca",
                "title": "Tome 12",
                "number": 12,
                "date_parution": "2012-10-19",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51qdqdxDqZL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505015475",
                "asin": "2505015473"
              },
              {
                "id": "5fceded4-b1c8-44ad-90c2-2512e2347669",
                "title": "Tome 11",
                "number": 11,
                "date_parution": "2012-07-06",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51w36HrDDhL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505014799",
                "asin": "2505014795"
              },
              {
                "id": "618faa60-f9f4-495e-ae04-6b232c560383",
                "title": "Tome 10",
                "number": 10,
                "date_parution": "2012-04-27",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51Jafyn2XUL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505014379",
                "asin": "250501437X"
              },
              {
                "id": "bc9c3b30-c11d-4495-98de-c3bf5e16e789",
                "title": "Tome 9",
                "number": 9,
                "date_parution": "2012-01-06",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/615UvI3NaWL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505049685",
                "asin": "2505049688"
              },
              {
                "id": "4e2bfa92-e111-4fca-a76d-2aed84deacc1",
                "title": "Tome 8",
                "number": 8,
                "date_parution": "2011-10-07",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51iV%2BuoYPcL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505012412",
                "asin": "2505012415"
              },
              {
                "id": "dec4d105-ed98-42a1-aefb-62d772b1d586",
                "title": "Tome 7",
                "number": 7,
                "date_parution": "2011-07-01",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51GHBMNri8L.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505011699",
                "asin": "2505011699"
              },
              {
                "id": "896c1b28-9d31-4f12-a307-b0e86b18ce34",
                "title": "Tome 6",
                "number": 6,
                "date_parution": "2011-04-01",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51xox2vAXYL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505010760",
                "asin": "2505010765"
              },
              {
                "id": "ae3c90d6-3a73-415e-8b3e-02d48910db9f",
                "title": "Tome 5",
                "number": 5,
                "date_parution": "2011-01-07",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51FEUifFO0L.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505010302",
                "asin": "2505010307"
              },
              {
                "id": "2e1afcfe-1e70-4f15-bba8-1acc089edd40",
                "title": "Tome 4",
                "number": 4,
                "date_parution": "2010-11-05",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51ihSKjlJXL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505009887",
                "asin": "2505009880"
              },
              {
                "id": "d6f8a236-4a04-4999-a8d1-d0f8c95c9a1f",
                "title": "Tome 3",
                "number": 3,
                "date_parution": "2010-09-17",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51UizLCakoL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505009634",
                "asin": "2505009635"
              },
              {
                "id": "87262606-aa57-4035-a83f-3baac10bf4e9",
                "title": "Tome 2",
                "number": 2,
                "date_parution": "2010-07-02",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51n4kKZnMSL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505008279",
                "asin": "2505008272"
              },
              {
                "id": "ec23f826-05bc-4bfc-b3e1-b2227e782ce9",
                "title": "Tome 1",
                "number": 1,
                "date_parution": "2010-07-02",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51ihjP9KCzL.jpg",
                "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
                "isbn": "9782505008262",
                "asin": "2505008264"
              }
            ]
          }
        ]
      }
    },
    {
      "id": "63698905-4149-434a-abfd-59e2343c1c07",
      "series_id": "320071be-4196-402b-98d7-d34bec8a1aab",
      "author_id": "17491b54-580f-4d5f-94e5-9b9e501871c4",
      "job_id": "bf39f3fe-ce75-48b0-b557-877da935633b",
      "series": {
        "id": "320071be-4196-402b-98d7-d34bec8a1aab",
        "title": "Death Note",
        "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d",
        "editions": [
          {
            "id": "908612fd-1cc7-4780-aea4-63d3ac04c6bd",
            "title": "Black Edition",
            "series_id": "320071be-4196-402b-98d7-d34bec8a1aab",
            "publisher_id": "4c9547ff-2ef6-439a-80b8-ea705a385b76",
            "volumes": [
              {
                "id": "ec9d0282-78a1-491e-bd1e-a937b332c56f",
                "title": "Tome 6",
                "number": 6,
                "date_parution": "2011-05-06",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/41FBONjMi4L.jpg",
                "edition_id": "908612fd-1cc7-4780-aea4-63d3ac04c6bd",
                "isbn": "9782505010951",
                "asin": "2505010951"
              },
              {
                "id": "ef282203-2ef5-4972-9598-d365fe7ff09c",
                "title": "Tome 5",
                "number": 5,
                "date_parution": "2011-03-04",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/41%2BDT%2B3-rwL.jpg",
                "edition_id": "908612fd-1cc7-4780-aea4-63d3ac04c6bd",
                "isbn": "9782505010692",
                "asin": "2505010692"
              },
              {
                "id": "69fed8ee-a500-4bde-9ba2-458f639785d1",
                "title": "Tome 4",
                "number": 4,
                "date_parution": "2011-01-07",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/41lRggimfJL.jpg",
                "edition_id": "908612fd-1cc7-4780-aea4-63d3ac04c6bd",
                "isbn": "9782505010326",
                "asin": "2505010323"
              },
              {
                "id": "7ac3f560-bb02-4a9e-9e01-67cc2e2b9d32",
                "title": "Tome 3",
                "number": 3,
                "date_parution": "2010-11-05",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/41mLoqm6D0L.jpg",
                "edition_id": "908612fd-1cc7-4780-aea4-63d3ac04c6bd",
                "isbn": "9782505009986",
                "asin": "2505009988"
              },
              {
                "id": "3a03733c-1a5a-4673-9299-28e5cc82a8af",
                "title": "Tome 2",
                "number": 2,
                "date_parution": "2010-09-03",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51KMP3phtGL.jpg",
                "edition_id": "908612fd-1cc7-4780-aea4-63d3ac04c6bd",
                "isbn": "9782505009061",
                "asin": "2505009066"
              },
              {
                "id": "dbab2483-2d3c-4832-9993-b6efbd765787",
                "title": "Tome 1",
                "number": 1,
                "date_parution": "2010-07-02",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/413rEPunmPL.jpg",
                "edition_id": "908612fd-1cc7-4780-aea4-63d3ac04c6bd",
                "isbn": "9782505009054",
                "asin": "2505009058"
              }
            ]
          },
          {
            "id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
            "title": "Edition simple",
            "series_id": "320071be-4196-402b-98d7-d34bec8a1aab",
            "publisher_id": "4c9547ff-2ef6-439a-80b8-ea705a385b76",
            "volumes": [
              {
                "id": "081648ff-1b14-4af7-b8f0-9e97b0ba6804",
                "title": "Tome 13",
                "number": 13,
                "date_parution": "2009-09-18",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51JrRDR1ANL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505004523",
                "asin": "2505004528"
              },
              {
                "id": "11606764-7aee-4e33-9df1-e9dc5298eda8",
                "title": "Tome 12",
                "number": 12,
                "date_parution": "2008-10-02",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/61XFUTPtQdL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505004295",
                "asin": "2505004293"
              },
              {
                "id": "a4477f49-cff1-4913-bd39-68fefbc10479",
                "title": "Tome 11",
                "number": 11,
                "date_parution": "2008-07-04",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/618b1GXgGFL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505003694",
                "asin": "2505003696"
              },
              {
                "id": "c1995831-6447-43e9-869b-74f2cf93966d",
                "title": "Tome 10",
                "number": 10,
                "date_parution": "2008-06-06",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/61OEH0gsrIL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505003038",
                "asin": "2505003033"
              },
              {
                "id": "73f6690b-716b-4659-97f5-d66ce10acd8a",
                "title": "Tome 9",
                "number": 9,
                "date_parution": "2008-04-11",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/61jqfMbZ8VL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505002970",
                "asin": "2505002975"
              },
              {
                "id": "12b5dc0d-253a-41dd-bbee-e1f31e947eb1",
                "title": "Tome 8",
                "number": 8,
                "date_parution": "2008-02-15",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51ZlUQFdaPL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505002697",
                "asin": "250500269X"
              },
              {
                "id": "f0bc0f76-bddf-4af8-a589-0d877aae9235",
                "title": "Tome 7",
                "number": 7,
                "date_parution": "2007-12-07",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/61qgonxtZOL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505001997",
                "asin": "2505001995"
              },
              {
                "id": "a2280876-7082-48b5-bab3-34c4e5eab279",
                "title": "Tome 6",
                "number": 6,
                "date_parution": "2007-10-05",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51ccrw7ky1L.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505001812",
                "asin": "2505001812"
              },
              {
                "id": "b95ab092-9baf-4feb-b8f1-e79b449b9c54",
                "title": "Tome 5",
                "number": 5,
                "date_parution": "2007-08-24",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/61bXKQlOllL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505001614",
                "asin": "2505001618"
              },
              {
                "id": "e7dbccb5-4710-446b-8c08-d35fb8a1eaeb",
                "title": "Tome 4",
                "number": 4,
                "date_parution": "2007-05-31",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/611d8aPhqhL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505001065",
                "asin": "2505001065"
              },
              {
                "id": "fa7a1162-d9e3-462f-a367-8371a55f7c94",
                "title": "Tome 3",
                "number": 3,
                "date_parution": "2007-04-05",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/61fiEjVbUbL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505000792",
                "asin": "2505000794"
              },
              {
                "id": "11d1491e-22d4-49f1-b29f-78007493d4e6",
                "title": "Tome 2",
                "number": 2,
                "date_parution": "2007-02-01",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/61DduSmdKGL.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505000426",
                "asin": "2505000425"
              },
              {
                "id": "ff15ae7e-fcfe-4061-a27b-3b481e5f313c",
                "title": "Tome 1",
                "number": 1,
                "date_parution": "2007-01-18",
                "img": "https://images-eu.ssl-images-amazon.com/images/I/51eBp6XUd5L.jpg",
                "edition_id": "5482eba9-2773-40d8-82f7-1870b1b2a3e9",
                "isbn": "9782505000327",
                "asin": "2505000328"
              }
            ]
          }
        ]
      }
    }
  ]
}