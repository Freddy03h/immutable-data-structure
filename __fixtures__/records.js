import Immutable from 'immutable'

export const initialState = Immutable.Map({
  data: Immutable.Map(),
  relations: Immutable.Map(),
})

export const SerieRecord = Immutable.Record({
  id: null,
  title: null,
  type_id: null,
})
export const serieForeignKeys = ['type_id']

export const EditionRecord = Immutable.Record({
  id: null,
  title: null,
  publisher_id: null,
  series_id: null,
})
export const editionForeignKeys = ['publisher_id', 'series_id']
