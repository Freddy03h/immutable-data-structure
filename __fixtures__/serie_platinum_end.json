{
  "id": "fc34ad95-27cf-4cba-aafc-e9c37de532b4",
  "title": "Platinum End",
  "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d",
  "possessions_count": 1024,
  "tasks": [
    {
      "id": "4b1d117d-063b-4a08-b6d7-2f3d252b2a81",
      "series_id": "fc34ad95-27cf-4cba-aafc-e9c37de532b4",
      "author_id": "f06df71d-7bb3-4a4c-aec0-638f53aa43a8",
      "job_id": "37fb2477-ec0d-4ffc-9d3b-4189c2d01629",
      "author": {
        "id": "f06df71d-7bb3-4a4c-aec0-638f53aa43a8",
        "name": "Obata",
        "first_name": "Takeshi"
      }
    },
    {
      "id": "5195c5d1-9d3e-4a6c-9b29-077fc217d224",
      "series_id": "fc34ad95-27cf-4cba-aafc-e9c37de532b4",
      "author_id": "17491b54-580f-4d5f-94e5-9b9e501871c4",
      "job_id": "bf39f3fe-ce75-48b0-b557-877da935633b",
      "author": {
        "id": "17491b54-580f-4d5f-94e5-9b9e501871c4",
        "name": "Ōba",
        "first_name": "Tsugumi"
      }
    }
  ],
  "editions": [
    {
      "id": "73173187-5965-451c-adbe-84e3e719abc0",
      "title": "Edition simple",
      "series_id": "fc34ad95-27cf-4cba-aafc-e9c37de532b4",
      "publisher_id": "122e0ec3-f072-4c5d-b40d-3eba4a82fe1e",
      "volumes": [
        {
          "id": "27e50b0a-785e-4298-af2f-ae492c57c8a6",
          "title": "Tome 6",
          "number": 6,
          "date": "2017-09-20",
          "img": "",
          "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
          "isbn": "9782820329011"
        },
        {
          "id": "f6f21458-e707-40ba-b9ec-f7f28d7c6343",
          "title": "Tome 5",
          "number": 5,
          "date": "2017-05-31",
          "img": "https://images-eu.ssl-images-amazon.com/images/I/61CV%2BQwDT7L.jpg",
          "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
          "isbn": "9782820328533"
        },
        {
          "id": "06fe038b-8c0d-4767-9167-ae3ca9bc3067",
          "title": "Tome 4",
          "number": 4,
          "date": "2017-03-08",
          "img": "https://images-eu.ssl-images-amazon.com/images/I/51tKK5n%2B-3L.jpg",
          "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
          "isbn": "9782820328175"
        },
        {
          "id": "7c0ad1a9-4a80-4710-b11b-eed249456924",
          "title": "Tome 3",
          "number": 3,
          "date": "2016-11-23",
          "img": "https://images-eu.ssl-images-amazon.com/images/I/51jAjenb0EL.jpg",
          "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
          "isbn": "9782820325402"
        },
        {
          "id": "975e5e39-8a47-4e0c-9a60-4a6256163257",
          "title": "Tome 2",
          "number": 2,
          "date": "2016-09-07",
          "img": "https://images-eu.ssl-images-amazon.com/images/I/51sgTWp8W-L.jpg",
          "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
          "isbn": "9782820324948"
        },
        {
          "id": "04754705-044e-4292-8ed2-eee50ee91d2d",
          "title": "Tome 1",
          "number": 1,
          "date": "2016-05-25",
          "img": "https://images-eu.ssl-images-amazon.com/images/I/51TgELYk1WL.jpg",
          "edition_id": "73173187-5965-451c-adbe-84e3e719abc0",
          "isbn": "9782820324603"
        }
      ]
    }
  ],
  "kinds": [
    {
      "id": "d5a3385a-cfcd-45f6-8f82-0c80956b084d",
      "title": "Shōnen"
    },
    {
      "id": "9d6a131c-4b4e-4cf6-a9aa-de78087d672d",
      "title": "Fantastique"
    },
    {
      "id": "acd6bec9-507d-46ab-a1de-f18f8b8ca83c",
      "title": "Drame"
    }
  ]
}