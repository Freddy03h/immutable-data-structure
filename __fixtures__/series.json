[
  {
    "id": "a02cf154-af6c-4f08-9a7a-32f7bc229ac8",
    "title": "One Piece",
    "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d"
  },
  {
    "id": "d1a38f41-2ac9-40cf-aa17-aeac9697e1c8",
    "title": "Naruto",
    "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d"
  },
  {
    "id": "96138837-ef12-4a8f-869a-ea745047b41f",
    "title": "Dreamland",
    "type_id": "3c009f18-811d-4b07-8dde-8249e422ec9e"
  },
  {
    "id": "940629b6-a0df-4bd2-970d-b75d19336793",
    "title": "Dragon Ball",
    "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d"
  },
  {
    "id": "f41a928c-8450-40e9-805b-3ccf3ac49705",
    "title": "Berserk",
    "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d"
  },
  {
    "id": "b9652346-95db-45ef-982b-de9f9d3ac01f",
    "title": "Fullmetal Alchemist",
    "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d"
  },
  {
    "id": "ed7eaace-939f-4c1c-ad13-8e6c336257cf",
    "title": "L'Attaque des Titans",
    "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d"
  },
  {
    "id": "320071be-4196-402b-98d7-d34bec8a1aab",
    "title": "Death Note",
    "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d"
  },
  {
    "id": "b54ddfa3-3188-4127-83aa-87917e8ec422",
    "title": "Soul Eater",
    "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d"
  },
  {
    "id": "f59ab8de-2c3a-4bf2-a501-b844599646be",
    "title": "Radiant",
    "type_id": "3c009f18-811d-4b07-8dde-8249e422ec9e"
  },
  {
    "id": "a2e0ddcf-71c6-406c-aadc-ccbac2d3f668",
    "title": "Hunter X Hunter",
    "type_id": "106f524e-7283-44b8-aa84-25e9a7fb1f7d"
  },
  {
    "id": "f0db2df6-dc0c-48cd-b3c4-88dd350c9cb0",
    "title": "Naruto : Les Romans",
    "type_id": "bb3d92a9-020a-4d3f-adec-77bebe9980db"
  }
]