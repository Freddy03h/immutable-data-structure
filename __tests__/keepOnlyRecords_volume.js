import Immutable from 'immutable'
import {
  initialState,
  createMergeRecords, createMergeCompleteListsRecords,
  keepOnlyRecords,
} from '../src/index'

import serieDeathNoteJSON from '../__fixtures__/serie_death_note.json'
import serieFullmetalAlchemistJSON from '../__fixtures__/serie_fullmetal_alchemist.json'
import authorObaJSON from '../__fixtures__/author_oba.json'
import { VolumeRecord, volumeForeignKeys } from '../__fixtures__/records'

const mergeVolumeRecords = createMergeRecords(VolumeRecord, volumeForeignKeys)
const mergeVolumeRecordsFromSeries = createMergeCompleteListsRecords(mergeVolumeRecords, 'volumes', 'edition_id')

const serie1 = Immutable.fromJS(serieDeathNoteJSON)
const serie2 = Immutable.fromJS(serieFullmetalAlchemistJSON)
const author1 = Immutable.fromJS(authorObaJSON)

describe('keepOnlyRecords volume', () => {

  test('big store', () => {
    const store1 = mergeVolumeRecordsFromSeries(initialState, serie1.get('editions'))
    const store2 = mergeVolumeRecordsFromSeries(store1, serie2.get('editions'))
    const editionsAuthor = author1.get('tasks')
      .map((task) => task.getIn(['series', 'editions']))
      .flatten(true)
    const store3 = mergeVolumeRecordsFromSeries(store2, editionsAuthor)

    const idsToKeep = Immutable.Set([
      'ec23f826-05bc-4bfc-b3e1-b2227e782ce9',
      '8d5329c8-2ad5-4713-8961-84f2e42a61c6',
      '58715dcb-ce62-4b8c-a6b0-6412310b12ff',
    ])

    expect(
      keepOnlyRecords(store3, idsToKeep)
    ).toMatchSnapshot()
  })

})
