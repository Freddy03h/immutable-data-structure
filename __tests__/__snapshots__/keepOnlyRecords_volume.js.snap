// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keepOnlyRecords volume big store 1`] = `
Immutable.Map {
  data: Immutable.Map {
    ec23f826-05bc-4bfc-b3e1-b2227e782ce9: Object {
      "content": null,
      "date": "2010-07-02",
      "edition_id": "8413f439-9bdd-42a2-bd40-4b0f6118dc8d",
      "id": "ec23f826-05bc-4bfc-b3e1-b2227e782ce9",
      "img": "https://images-eu.ssl-images-amazon.com/images/I/51ihjP9KCzL.jpg",
      "isbn": "9782505008262",
      "number": 1,
      "title": "Tome 1",
    },
    8d5329c8-2ad5-4713-8961-84f2e42a61c6: Object {
      "content": null,
      "date": "2012-03-08",
      "edition_id": "d42c9391-c92f-4485-a772-8df06d70ed17",
      "id": "8d5329c8-2ad5-4713-8961-84f2e42a61c6",
      "img": "https://images-eu.ssl-images-amazon.com/images/I/51MOHY8mkuL.jpg",
      "isbn": "9782351427552",
      "number": 1,
      "title": "Tome I (1-2-3)",
    },
    58715dcb-ce62-4b8c-a6b0-6412310b12ff: Object {
      "content": null,
      "date": "2006-11-09",
      "edition_id": "ffdf56ae-058d-49c6-80c6-336471a73dcc",
      "id": "58715dcb-ce62-4b8c-a6b0-6412310b12ff",
      "img": "https://images-eu.ssl-images-amazon.com/images/I/51N7A347E8L.jpg",
      "isbn": "9782351420492",
      "number": 9,
      "title": "Tome 9",
    },
  },
  relations: Immutable.Map {
    edition_id: Immutable.Map {
      d42c9391-c92f-4485-a772-8df06d70ed17: Immutable.OrderedSet [
        "8d5329c8-2ad5-4713-8961-84f2e42a61c6",
      ],
      ffdf56ae-058d-49c6-80c6-336471a73dcc: Immutable.OrderedSet [
        "58715dcb-ce62-4b8c-a6b0-6412310b12ff",
      ],
      8413f439-9bdd-42a2-bd40-4b0f6118dc8d: Immutable.OrderedSet [
        "ec23f826-05bc-4bfc-b3e1-b2227e782ce9",
      ],
    },
  },
}
`;
